### PR TITLE
Display a more helpful error message for a removed user

### PR DESCRIPTION
**Pull request by ehuss**
The current error message when a user is deleted is somewhat difficult to deal with:

```
[ERROR rust_team::validate] validation error: couldn't verify GitHub usernames: graphql error: Could not resolve to a node with the global id of 'MDQ6VXNlcjMzOTgzMzMy'.
```

The solution here is to go through the batch of 100 user IDs and query each one until it fails to determine which GitHub user was deleted.

### CODE REVIEW

**Review by rylev — APPROVED** [review]

File: src/github.rs [hunk 1]
```diff
@@ -136,14 +136,49 @@ impl GitHubApi {
...
                     ids: chunk.iter().map(|id| user_node_id(*id)).collect(),
                 },
-            )?;
+            ) {
+                Ok(res) => res,
+                Err(e) => {
+                    if cant_resolve(&e) {
+                        // This error happens when a user is deleted. Provide
+                        // a more helpful error message to pinpoint it:
+                        for id in chunk {
```
Comment (rylev) [inline src/github.rs:154]: Is doing a binary search worth the potential added complexity? Doing potentially 100 requests for the error message sounds like quite a lot...

**Comment by rylev** [comment]
A rebase should fix the CI issue.

**Review by rylev — APPROVED** [review]

File: src/github.rs (hunk 1 above)
Comment (rylev) [reply src/github.rs:154]: Yes exactly. Binary search would definitely help reduce the number of requests. I agree we're probably ok just making the possible ~100 requests, but we *might* run into issues in the future. CI runs every hour, and I've also considered making it every 10 minutes (to make sure things are always in sync).

Comment (ehuss) [reply src/github.rs:154]: Ah, I didn't know about the scheduled jobs.

Now that I think about it, I'm not actually sure what the limit is here. I think the rust-lang org is an Enterprise account, which I think has a limit of 15,000 requests per hour per repo using the GitHub Actions auto-generated token.

**Pull request merged**